src/ui/public/courier/data_source/search_source.js

```diff
@@ -288,5 +292,16 @@ export function SearchSourceProvider(Promise, Private, config) {
     }
   };
 
+  SearchSource.prototype.clone = function () {
+    const clone = new SearchSource(this.toString());
+    clone.set('index', this.get('index'));
+    clone.inherits(this.getParent());
+    return clone;
+  };
+
+  SearchSource.prototype.getESQuery = function () {
+    return this._flatten();
```

Comment:
In `getSharingData`, could you get the index from the searchSource separately like this?

```
    return {
      esQuery,
      index: searchSource.get('index'),
      fields: selectFields,
      metaFields: $scope.indexPattern.metaFields,
      conflictedTypesFields: $scope.indexPattern.fields.filter(f => f.type === 'conflict').map(f => f.name),
    };
```

I just think the return value from _flatten is sharing too much of the searchSource's internal state.